### PR TITLE
fix testeth --createRandomTests + expectException in blockchain tests

### PR DESCRIPTION
**Pull request by winsvega**
- Blockchain tests expectedException section
- restore create random tests in testeth (fix)
- Unknown Parent exception info

### CODE REVIEW

File: test/TestHelper.cpp [hunk 1]
```diff
@@ -813,6 +810,15 @@ Options::Options()
 		}
 		else if (arg == "--createRandomTest")
 			createRandomTest = true;
+		else if (arg == "-t" && i + 1 < argc)
+			rCurrentTestSuite = std::string{argv[i + 1]};
+		else if (arg == "-checktest" || arg == "-filltest")
```
Comment (CJentzsch) [inline test/TestHelper.cpp:824]: use `-` for single letters and `--` for longer words.

File: test/TestHelper.cpp [hunk 2]
```diff
@@ -813,6 +810,15 @@ Options::Options()
...
+		{
+			//read all line to the end
+			for (int j = i+1; j < argc; ++j)
```
Comment (CJentzsch) [inline test/TestHelper.cpp:827]: `i + 1` instead of `i+1`

File: test/boostTest.cpp [hunk 3]
```diff
@@ -28,55 +28,90 @@
...
 #pragma GCC diagnostic pop
+
+#include <stdlib.h>
 #include <test/TestHelper.h>
 
 using namespace boost::unit_test;
 
-std::vector<char*> originalArgs;
+int gArgc;
+char** gArgv;
```
Comment (CJentzsch) [inline test/boostTest.cpp]: why don't you use a vector?

File: test/boostTest.cpp [hunk 4]
```diff
@@ -28,55 +28,90 @@
...
+		for (int i = 0; i < argc; i++)
+		{
+			std::string arg = std::string{argv[i]};
+
+			//replace test suite to random tests
+			if (arg == "-t" && i+1 < argc)
+				argv[i+1] = (char*)std::string("RandomTestCreationSuite").c_str();
+
+			//don't pass long raw test input to boost
+			if (arg == "-checktest")
```
Comment (CJentzsch) [inline test/boostTest.cpp]: `--`

File: test/boostTest.cpp [hunk 5]
```diff
@@ -28,55 +28,90 @@
...
+			{
+				argc = i+1;
```
Comment (CJentzsch) [inline test/boostTest.cpp]: spaces

File: test/boostTest.cpp [hunk 6]
```diff
@@ -28,55 +28,87 @@
...
 #include <boost/test/included/unit_test.hpp>
 #pragma GCC diagnostic pop
+
+#include <stdlib.h>
 #include <test/TestHelper.h>
 
 using namespace boost::unit_test;
 
-std::vector<char*> originalArgs;
+std::vector<char*> parametrs;
```
Comment (CJentzsch) [inline test/boostTest.cpp]: typo

**Comment by gavofyork** [comment]
conflicts.

**Comment by CJentzsch** [comment]
All builds fail, please check the reason, or retrigger the build.

File: test/TestHelper.cpp [hunk 7]
```diff
@@ -716,11 +716,8 @@ RLPStream createRLPStreamFromTransactionFields(json_spirit::mObject const& _tObj
 	return rlpStream;
 }
 
-Options::Options()
-{
-	auto argc = boost::unit_test::framework::master_test_suite().argc;
-	auto argv = boost::unit_test::framework::master_test_suite().argv;
-
+Options::Options(int argc, char** argv)
```
Comment (CJentzsch) [inline test/TestHelper.cpp:728]: forgot the underscore

File: test/TestHelper.cpp [hunk 8]
```diff
@@ -820,9 +826,9 @@ Options::Options()
 		g_logVerbosity = -1;	//disable cnote but leave cerr and cout
 }
 
-Options const& Options::get()
+Options const& Options::get(int argc, char** argv)
```
Comment (CJentzsch) [inline test/TestHelper.cpp:838]: underscore

**Pull request closed without merging**